src/Draco.LanguageServer/Capabilities/TextDocumentSync.cs

```diff
@@ -29,11 +30,16 @@ private async Task PublishDiagnosticsAsync(DocumentUri uri)
         var compilation = this.compilation;
 
         var syntaxTree = GetSyntaxTree(compilation, uri);
-        if (syntaxTree is null) return;
 
-        var semanticModel = compilation.GetSemanticModel(syntaxTree);
-        var diags = semanticModel.Diagnostics;
-        var lspDiags = diags.Select(Translator.ToLsp).ToList();
+        var lspDiags = new List<Diagnostic>();
+        // If the tree is null we want to push empty diags
+        if (syntaxTree is not null)
+        {
+            var semanticModel = compilation.GetSemanticModel(syntaxTree);
+            var diags = semanticModel.Diagnostics;
+            lspDiags = diags.Select(Translator.ToLsp).ToList();
+        }
```

Comment:
I'd invert the condition to reflect the original and publish empty in there, as it's a "sad path"